modules/meta/arrow-meta/src/main/java/arrow/generic/CoproductFileGenerator.kt

```diff
@@ -171,6 +220,28 @@ private fun FileSpec.Builder.addFoldFunction(generics: List<String>): Unit {
     )
 }
 
+private fun methodDocumentation(
+        output: String,
+        description: String,
+        input: List<Pair<String, String>> = emptyList()
+): String {
+    val parameters = if (input.isEmpty()) {
+        ""
+    } else {
+        input.joinToString(separator = "") { "@param ${it.first} ${it.second}\n" } + "\n"
+    }
+
+    return "$description\n\n$parameters@return $output\n"
+}
+
+private fun String.toCamelCase(): String {
```

Comment:
Nothing wrong with this implementation style but soon we will enforce across all arrow modules a style that is more succinct and expression oriented:
```kotlin
private fun String.toCamelCase(): String =
  if (isEmpty()) this 
  else first().toLowerCase() + substring(1)
```
Also this function may be public and placed in the arrow meta String utils file.